core/src/main/scala/za/co/absa/spline/harvester/conf/DefaultSplineConfigurer.scala

```diff
@@ -94,35 +96,55 @@ class DefaultSplineConfigurer(sparkSession: SparkSession, userConfiguration: Con
   override def queryExecutionEventHandler: QueryExecutionEventHandler =
     new QueryExecutionEventHandler(harvesterFactory, lineageDispatcher)
 
-  protected def lineageDispatcher: LineageDispatcher = instantiate[LineageDispatcher](
-    configuration.getRequiredString(LineageDispatcherClass))
-
-  protected def ignoredWriteDetectionStrategy: IgnoredWriteDetectionStrategy = instantiate[IgnoredWriteDetectionStrategy](
-    configuration.getRequiredString(IgnoreWriteDetectionStrategyClass))
+  protected def lineageDispatcher: LineageDispatcher =
+    configuration
+      .getOptionalString(DeprecatedConfProperty.LineageDispatcherClass)
+      .map(instantiate[LineageDispatcher])
+      .getOrElse{
```

Comment:
Shouldn't it be in the opposite way - read the new property first, and if one isn't found then try the deprecated one?
Instead of dealing with deprecated properties directly in the same place where actual properties are dealt with, I'd suggest to try to isolate them in the dedicated `Configuration` adapter, and then attach it to the configuration chain right before the `default.properties`.

E.g.
```scala
private lazy val configuration = new CompositeConfiguration(Seq(
    userConfiguration,
    new Spline05ConfigurationAdapter(userConfiguration), <--- This one here!
    new PropertiesConfiguration(defaultPropertiesFileName)
  ).asJava)
```
`Spline05ConfigurationAdapter` would translate the new properties to the old ones and vice versa.
This way the original config stack would be re-tried for the old properties, and if not found the `default.properties` would be used for the new ones.

The backward compatibility logic would be 100% encapsulated in a dedicated class, better testable, leaving the rest of config logic clean.